src/System.Console/tests/TermInfo.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using Xunit;
+
+public class TermInfo
+{
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyInstalledTermInfosParse()
+    {
+        foreach (string location in ConsolePal.TermInfo.Database._terminfoLocations)
+        {
+            if (!Directory.Exists(location))
+                continue;
+
+            foreach (string term in Directory.EnumerateFiles(location, "*", SearchOption.AllDirectories))
+            {
+                if (term.ToUpper().Contains("README")) continue;
+                ConsolePal.TerminalColorInfo info = new ConsolePal.TerminalColorInfo(ConsolePal.TermInfo.Database.ReadDatabase(Path.GetFileName(term)));
+
+                if (!string.IsNullOrEmpty(info.ForegroundFormat))
+                {
+                    Assert.NotEmpty(ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.ForegroundFormat, 0 /* irrelevant, just an integer to put into the formatting*/));
+                }
+
+                if (!string.IsNullOrEmpty(info.BackgroundFormat))
+                {
+                    Assert.NotEmpty(ConsolePal.TermInfo.ParameterizedStrings.Evaluate(info.ForegroundFormat, 0 /* irrelevant, just an integer to put into the formatting*/));
+                }
+            }
+        }
+    }
+
+    [Theory]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00331m", "\u001B\u005B\u00341m", 1)]
+    [InlineData("screen", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("screen", "\u001B\u005B\u00332m", "\u001B\u005B\u00342m", 2)]
+    [InlineData("Eterm", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("Eterm", "\u001B\u005B\u00333m", "\u001B\u005B\u00343m", 3)]
+    [InlineData("wsvt25", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("wsvt25", "\u001B\u005B\u00334m", "\u001B\u005B\u00344m", 4)]
+    [InlineData("mach-color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("mach-color", "\u001B\u005B\u00335m", "\u001B\u005B\u00345m", 5)]
```

Comment:
Nit: Could you also add some that use a value in the range [8,15] ?  The color values are broken into two groups: dark colors in the range [0,7] and light colors in the range [8,15], and some of these strings have special conditional logic for evaluating the cases where the values are greater than 7... I'd like to make sure we handle those correctly.